src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -888,7 +1074,15 @@ public MultiConnectSocketAsyncEventArgs() : base(unsafeSuppressExecutionContextF
             public short Version => _mrvtsc.Version;
             public void Reset() => _mrvtsc.Reset();
 
-            protected override void OnCompleted(SocketAsyncEventArgs e) => _mrvtsc.SetResult(true);
+            protected override void OnCompleted(SocketAsyncEventArgs e)
+            {
+                try
+                {
+                    // we dont't have TrySetResult and there could be two completing connections
+                    _mrvtsc.SetResult(true);
```

Comment:
Can't this be avoided by getting rid of the `internalArgs.SecondarySaea.Completed` subscription?

Reply:
I was not sure if that is safe when the operations are racing. I would suggest to keep it end refactor it if we get more confidence.